src/server/server.ts

```diff
@@ -55,6 +56,40 @@ app.post('/api/DataPlane', (req: express.Request, res: express.Response) => {
     }
 });
 
+app.post('/api/CloudToDevice', (req: express.Request, res: express.Response) => {
+    try {
+        if (!req.body) {
+            res.status(BAD_REQUEST).send();
+        }
+        else {
+            const hubClient = HubClient.fromConnectionString(req.body.connectionString);
+            // tslint:disable-next-line:cyclomatic-complexity
+            hubClient.open(() => {
+                const message = new CloudToDeviceMessage(req.body.body);
+                // add properties to message
+                if (req.body.properties && req.body.properties.length !== 0) {
```

Comment:
Try extracting this to a new method `getC2DMessageProperties(req.body.properties)` to remove the need to break cyclomatic complexity.